src/CodeGen_LLVM.cpp

```diff
@@ -3156,6 +3156,37 @@ void CodeGen_LLVM::visit(const Call *op) {
         builder->setDefaultFPMathTag(strict_fp_math_md);
 
         value = builder->CreateFCmpUNO(a, a);
+    } else if (op->call_type == Call::PureExtern &&
+               (op->name == "is_inf_f32" || op->name == "is_inf_f64")) {
+        internal_assert(op->args.size() == 1);
+
+        IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>::FastMathFlagGuard guard(*builder);
+        llvm::FastMathFlags safe_flags;
+        safe_flags.clear();
+        builder->setFastMathFlags(safe_flags);
+        builder->setDefaultFPMathTag(strict_fp_math_md);
+
+        // isinf(e) -> (fabs(e) == infinity)
+        Expr e = op->args[0];
+        internal_assert(e.type().is_float());
+        Expr inf = e.type().max();
+        codegen(abs(e) == inf);
```

Comment:
would codegen(strict_float(abs(e) == inf)) be equivalent but shorter?

Reply:
That's a good question -- I was following the pattern used by is_nan here. Maybe. Though I think a "righter" answer is probably to follow this with a fix for https://github.com/halide/Halide/issues/4281 (which I filed this morning), as the use of these calls outside a strict_float context in the first place is likely an error on the coder's part anyway.